jenkins/spark-nightly-build.sh

```diff
@@ -21,6 +21,8 @@ set -ex
 
 ## export 'M2DIR' so that shims can get the correct Spark dependency info
 export M2DIR=${M2DIR:-"$WORKSPACE/.m2"}
+## export 'LC_ALL' to set locale with UTF-8 so regular expressions are enabled
+export LC_ALL="en_US.UTF-8"
```

Comment:
this config is for all tests (nightly in this case, and also the update in premerge script)

one question is,
are we going to enable UTF-8 for all our integrations tests?
if yes, we should doc this somewhere (like https://github.com/NVIDIA/spark-rapids/tree/branch-22.08/integration_tests#prerequisities)to tell all our developers that our IT require UTF-8 as the context.
if no and only regex text need it, this should only be set as specific test-level context (inside the test file) instead of for all tests in script IMHO

Reply:
Hmm, I would think (if not now then eventually) string integration tests (non-regular expression tests) might require UTF-8 support as well. For now, they don't seem to have this depedency.

I have noticed this pattern in the premerge script, but did not see an equivalent in the nightly script:

```
INCLUDE_SPARK_AVRO_JAR=true TEST='avro_test.py' ./integration_tests/run_pyspark_from_build.sh
```

Regular expressions have 2 sets of tests, unit tests (Scalatest) and integration tests (pytest).  The Unicode-specific unit tests are skipped when UTF-8 is not in the environment (and therefore would run and pass), but the functionality itself would only be enabled when UTF-8 is in the environment (so these tests are testing an invalid environment in that case).  The integration tests in `regexp_test.py` could handled similarly to the above pattern (which is also used for for `regexp_no_unicode_test.py`). 

```
LC_ALL="en_US.UTF-8" TEST="regexp_test.py" ./integration_tests/run_pyspark_from_build.sh
```

This case though is only done in the premerge script. How would this be handled in the nightly script? And how would we handle the scala tests?

Reply:
> How would this be handled in the nightly script? And how would we handle the scala tests?

We could have multiple ways to do so like separate the CI in different stages/jobs, like use ENV/Arg to control only run specific tests. But I would like to know more inputs if below is already the plan cc @jlowe @revans2 @tgravescs 
```
Hmm, I would think (if not now then eventually) string integration tests (non-regular expression tests) might require UTF-8 support as well.
```